### PR TITLE
feat: add ttlSecondsAfterFinished to setup and cleanup jobs

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Add ttlSecondsAfterFinished to setup and cleanup jobs to ensure that the job is removed after installation or cleanup

### Checklist

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -153,6 +153,9 @@ sumologic:
       ## Add custom annotations only to setup job pod
       podAnnotations: {}
 
+      ## Time in seconds after which the job result will be removed from the Kubernetes cluster
+      ttlSecondsAfterFinished: 0
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:157]: Should this really be 0 by default? We should leave enough time for the user to be able to check the output if the Job failed. I'd put it at at least 60, maybe 120 to be on the safe side.

File: tests/helm/testdata/goldenfile/cleanup/sumologic-mock-no-secret.output.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ metadata:
     release: "RELEASE-NAME"
     heritage: "Helm"
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/cleanup/sumologic-mock-no-secret.output.yaml:18]:
```suggestion
  ttlSecondsAfterFinished: 120
```

File: tests/helm/testdata/goldenfile/cleanup/sumologic-mock.output.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@ metadata:
     release: "RELEASE-NAME"
     heritage: "Helm"
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/cleanup/sumologic-mock.output.yaml:18]:
```suggestion
  ttlSecondsAfterFinished: 120
```

File: tests/helm/testdata/goldenfile/setup/basic.output.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ metadata:
     release: "RELEASE-NAME"
     heritage: "Helm"
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/setup/basic.output.yaml:18]:
```suggestion
  ttlSecondsAfterFinished: 120
```

File: tests/helm/testdata/goldenfile/setup/sumologic-mock-no-secret.output.yaml [hunk 5]
```diff
@@ -15,6 +15,7 @@ metadata:
     release: "RELEASE-NAME"
     heritage: "Helm"
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/setup/sumologic-mock-no-secret.output.yaml:18]:
```suggestion
  ttlSecondsAfterFinished: 120
```

File: tests/helm/testdata/goldenfile/setup/sumologic-mock.output.yaml [hunk 6]
```diff
@@ -15,6 +15,7 @@ metadata:
     release: "RELEASE-NAME"
     heritage: "Helm"
 spec:
+  ttlSecondsAfterFinished: 0
```
Comment (sumo-drosiek) [inline tests/helm/testdata/goldenfile/setup/sumologic-mock.output.yaml:18]:
```suggestion
  ttlSecondsAfterFinished: 120
```

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml]: changed

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**